src/status_im/contexts/wallet/wallet_connect/events.cljs

```diff
@@ -36,7 +36,9 @@
 (rf/reg-event-fx
  :wallet-connect/on-session-proposal
  (fn [{:keys [db]} [proposal]]
-   {:db (assoc db :wallet-connect/current-proposal proposal)}))
+   {:db            (assoc db :wallet-connect/current-proposal proposal)
+    :open-modal-fx [:screen/wallet.wallet-connect-session-proposal
+                    (:theme db)]}))
```

Comment:
effects should go into the `:fx` key e.g. `{:db ... :fx [[:open-modal-fx [...]]]}`

Reply:
Also, I don't really see the effect being used directly in the codebase. Usually we're dispatching the `:open-modal` event, which does the same and also takes care of injecting the theme.

Reply:
Yeah, but in this case, we cannot dispatch the event (to open modal), because this event is not user generated. 

So I did what the `open-modal` event handler does internally.

Reply:
@shivekkhurana 
What @clauxx is referring to is to use the `:fx` effect to dispatch re-frame events

Reply:
I understand the `:fx` thing. My answer was for "Usually we're dispatching the :open-modal". We cannot `rf/dispatch` here, but I'll move `:open-modal-fx` to `:fx` key.

Reply:
> We cannot rf/dispatch here

I don't understand why we can't. Couldn't we just do 

```clj
{:db (...)
 :fx [[:dispatch [:open-modal :screen/wallet.wallet-connect-session-proposal]]]}
```

Reply:
> > We cannot rf/dispatch here
> 
> I don't understand why we can't. Couldn't we just do
> 
> ```clojure
> {:db (...)
>  :fx [[:dispatch [:open-modal :screen/wallet.wallet-connect-session-proposal]]]}
> ```

I understand now. I didn't know this is possible. I'll make the required change.